Source/AppleSocial.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  AppleSocial.swift
+//  edX
+//
+//  Created by Saeed Bashir on 8/24/20.
+//  Copyright © 2020 edX. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import MSAL
+import AuthenticationServices
+
+typealias AppleLoginCompletionHandler =  (OEXRegisteringUserDetails?, _ accessToken: String?, Error?) -> Void
+
+
+class AppleSocial: NSObject {
+
+    static let shared = AppleSocial()
+    var completionHandler: AppleLoginCompletionHandler!
+
+    private override init() {
+        super.init()
+    }
+
+    func loginFromController(controller : UIViewController, completion: @escaping AppleLoginCompletionHandler) {
+
+        completionHandler = completion
+
+        if #available(iOS 13.0, *) {
+            let authorizationProvider = ASAuthorizationAppleIDProvider()
+            let request = authorizationProvider.createRequest()
+            request.requestedScopes = [.fullName, .email]
+
+            let authorizationController = ASAuthorizationController(authorizationRequests: [request])
+            authorizationController.delegate = self
+            authorizationController.presentationContextProvider = self
+            authorizationController.performRequests()
+        }
+        else {
+            let error = NSError(domain: NSCocoaErrorDomain, code: -10000, userInfo: ["messgae": "Apple Sign In not supported prior to iOS 13"])
+            completionHandler(nil, nil, error)
+        }
+    }
+}
+
+@available(iOS 13.0, *)
+extension AppleSocial: ASAuthorizationControllerDelegate {
+
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithError error: Error) {
+        completionHandler(nil, nil, error)
+    }
+
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
+
+        guard let credentials = authorization.credential as? ASAuthorizationAppleIDCredential else {
+            let error = NSError(domain: NSCocoaErrorDomain, code: ASAuthorizationError.unknown.rawValue, userInfo: ["messgae": "Failed to get credentials"])
```

Comment:
instead of using `NSCocoaErrorDomain` please use class name as domain so callee function can have idea from where error was generated.

instead of using `messgae` as key, please use `NSLocalizedDescriptionKey` so when callee function prints or display information, the value will be available by using `error.localizedDescription`  

```suggestion
            let error = NSError(domain: "AppleSocial", code: ASAuthorizationError.unknown.rawValue, userInfo: [NSLocalizedDescriptionKey: "Failed to get credentials"])
```